kube-client/src/client/builder.rs

```diff
@@ -0,0 +1,181 @@
+use bytes::Bytes;
+use http::{Request, Response};
+use hyper::{self, client::HttpConnector};
+use hyper_timeout::TimeoutConnector;
+pub use kube_core::response::Status;
+use tower::{util::BoxService, BoxError, Layer, Service, ServiceBuilder};
+use tower_http::{
+    classify::ServerErrorsFailureClass, map_response_body::MapResponseBodyLayer, trace::TraceLayer,
+};
+
+use crate::{client::ConfigExt, Client, Config, Error, Result};
+
+/// HTTP body of a dynamic backing type.
+///
+/// The suggested implementation type is [`hyper::Body`].
+pub type DynBody = dyn http_body::Body<Data = Bytes, Error = BoxError> + Send + Unpin;
```

Comment:
Boxing the body should allow us to decouple from `hyper::Body` later on (which would be helpful for #848). This PR only tries to preserve `Body` agnosticism where it already exists, but in the longer term it may be worth investigating replacing most `hyper::Body` uses with `DynBody`.